File: SofiForce.Models/Models/EntityModels/SupervisorTypeModel.cs
using System;

namespace Models
{
    public class SupervisorTypeModel
    {
        public int SupervisorTypeId { get; set; }
        public string SupervisorTypeCode { get; set; }
        public string SupervisorTypeNameEn { get; set; }
        public string SupervisorTypeNameAr { get; set; }
        public bool? IsActive { get; set; }
        public bool? CanEdit { get; set; }
        public bool? CanDelete { get; set; }
        public int? DisplayOrder { get; set; }
        public string Color { get; set; }
        public string Icon { get; set; }
        public int? CBy { get; set; }
        public DateTime? CDate { get; set; }
        public int? EBy { get; set; }
        public DateTime? EDate { get; set; }
    }
}

File: SofiForce.Web/ClientApp/src/app/core/Models/EntityModels/SupervisorTypeModel.ts
import { SupervisorModel } from "./supervisorModel";

export interface SupervisorTypeModel {
    supervisorTypeId: number;
    supervisorTypeCode: string;
    supervisorTypeNameEn: string;
    supervisorTypeNameAr: string;
    isActive: boolean | null;
    canEdit: boolean | null;
    canDelete: boolean | null;
    displayOrder: number | null;
    color: string;
    icon: string;
    cBy: number | null;
    cDate: Date | null;
    eBy: number | null;
    eDate: Date | null;

}